website/server/libs/cron.js

```diff
@@ -461,20 +435,23 @@ export function cron (options = {}) {
 
   // Add 10 MP, or 10% of max MP if that'd be more. Perform this after Perfect Day for maximum benefit
   // Adjust for fraction of dailies completed
-  if (dailyDueUnchecked === 0 && dailyChecked === 0) dailyChecked = 1;
-  user.stats.mp += _.max([10, 0.1 * common.statsComputed(user).maxMP]) * dailyChecked / (dailyDueUnchecked + dailyChecked);
-  if (user.stats.mp > common.statsComputed(user).maxMP) user.stats.mp = common.statsComputed(user).maxMP;
+  if (!user.preferences.sleep) {
+    if (dailyDueUnchecked === 0 && dailyChecked === 0) dailyChecked = 1;
+    user.stats.mp += _.max([10, 0.1 * common.statsComputed(user).maxMP]) * dailyChecked / (dailyDueUnchecked + dailyChecked);
+    if (user.stats.mp > common.statsComputed(user).maxMP) user.stats.mp = common.statsComputed(user).maxMP;
+  }
 
   // After all is said and done, progress up user's effect on quest, return those values & reset the user's
-  let progress = user.party.quest.progress;
-  _progress = progress.toObject(); // clone the old progress object
-  progress.down = -1300;
-  _.merge(progress, {down: 0, up: 0, collectedItems: 0});
-
-  // Send notification for changes in HP and MP
+  if (!user.preferences.sleep) {
+    let progress = user.party.quest.progress;
+    _progress = progress.toObject(); // clone the old progress object
+    progress.down = -1300;
+    _.merge(progress, {down: 0, up: 0, collectedItems: 0});
```

Comment:
this is not specifically related to this PR but we're setting progress.down to -1300 and right after to 0?